src/app/components/navigation/create_playlist.rs

```diff
@@ -0,0 +1,72 @@
+use crate::app::components::HomeModel;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::CompositeTemplate;
+use std::rc::Rc;
+
+mod imp {
+    use super::*;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(resource = "/dev/alextren/Spot/components/create_playlist.ui")]
+    pub struct CreatePlaylistWidget {
+        #[template_child]
+        pub label: TemplateChild<gtk::Label>,
+
+        #[template_child]
+        pub entry: TemplateChild<gtk::Entry>,
+
+        #[template_child]
+        pub button: TemplateChild<gtk::Button>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for CreatePlaylistWidget {
+        const NAME: &'static str = "CreatePlaylistWidget";
+        type Type = super::CreatePlaylistWidget;
+        type ParentType = gtk::Popover;
+
+        fn class_init(klass: &mut Self::Class) {
+            Self::bind_template(klass);
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+
+    impl ObjectImpl for CreatePlaylistWidget {}
+    impl WidgetImpl for CreatePlaylistWidget {}
+    impl PopoverImpl for CreatePlaylistWidget {}
+}
+
+glib::wrapper! {
+    pub struct CreatePlaylistWidget(ObjectSubclass<imp::CreatePlaylistWidget>) @extends gtk::Widget, gtk::Popover;
+}
+
+impl CreatePlaylistWidget {
+    pub fn new(parent: &gtk::ListBoxRow, model: Rc<HomeModel>, user: Rc<String>) -> Self {
+        let w: CreatePlaylistWidget =
+            glib::Object::new(&[]).expect("Failed to create an instance of CreatePlaylistWidget");
+        w.connect_create(model, user);
+        w.set_parent(parent);
+        w.popup();
+        w
+    }
+
+    fn connect_create(&self, model: Rc<HomeModel>, user: Rc<String>) {
```

Comment:
i'd prefer to follow the usual pattern of passing an impl Fn here, so that the widget does not depend on some specific model, but rather just exposes a way to handle its activation

(also see other comments, user can be removed here)

Reply:
Helpful, and changed